reactivesocket-core/src/main/java/io/reactivesocket/Frame.java

```diff
@@ -397,6 +397,11 @@ public static int requestN(final Frame frame) {
     public static class Request {
         private Request() {}
 
+        public static Frame from(int streamId, FrameType type, Payload payload, long initialRequestN) {
+            int v = initialRequestN > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) initialRequestN;
```

Comment:
I added this as convenance method for the doOnRequest method. ReactiveStreams sends back-pressure signals as a Long. doOnRequest will only return positive numbers. I do the same thing in the Frame.RequestN.from method.